app/src/main/java/com/bashkevich/androidfundamentals/MainActivity.kt

```diff
@@ -1,38 +1,83 @@
 import androidx.work.*
+import com.bashkevich.androidfundamentals.model.database.MoviesDatabase
+import com.bashkevich.androidfundamentals.model.database.entity.MovieEntity
 import com.bashkevich.androidfundamentals.model.workmanager.MovieLoaderWorker
+import com.bashkevich.androidfundamentals.model.workmanager.MovieNotificationsWorker
+import com.bashkevich.androidfundamentals.model.workmanager.WorkerRepository
+import com.bashkevich.androidfundamentals.moviesdetails.view.FragmentMoviesDetails
 import com.bashkevich.androidfundamentals.movieslist.view.FragmentMoviesList
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import java.util.concurrent.TimeUnit
 
 class MainActivity : AppCompatActivity() {
+    private val workerRepository = WorkerRepository()
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
         if (savedInstanceState == null) {
             supportFragmentManager.commit {
-                setReorderingAllowed(true)
-                add<FragmentMoviesList>(R.id.main_container)
+                replace(R.id.main_container, FragmentMoviesList())
+            }
+            intent?.let(::handleMovieIntent)
+
+            lifecycleScope.launch {
+
+                WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
+                    MovieLoaderWorker.UPDATE_MOVIES,
+                    ExistingPeriodicWorkPolicy.REPLACE,
+                    workerRepository.periodicMovieRequest
+                )
+                WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
+                    MovieNotificationsWorker.NOTIFY,
+                    ExistingPeriodicWorkPolicy.REPLACE,
+                    workerRepository.periodicNotificationRequest
+                )
             }
         }
+    }
 
-        lifecycleScope.launchWhenCreated {
-            val constraints =
-                Constraints.Builder().setRequiredNetworkType(NetworkType.UNMETERED).build()
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        if (intent != null) {
+            handleMovieIntent(intent)
+        }
+    }
 
-            val periodicMovieRequest =
-                PeriodicWorkRequest.Builder(MovieLoaderWorker::class.java, 8, TimeUnit.HOURS)
-                    .setConstraints(constraints)
-                    .build()
+    private fun handleMovieIntent(intent: Intent?) {
+        when (intent?.action) {
+            Intent.ACTION_VIEW -> {
+                val id = intent.data?.lastPathSegment?.toIntOrNull()
+                if (id != null) {
+                    openMovie(id)
+                }
+            }
+        }
+    }
 
-            WorkManager.getInstance(applicationContext).enqueue(periodicMovieRequest)
+    private fun openMovie(id: Int) {
+        supportFragmentManager.popBackStack(
```

Comment:
Why pop?

Reply:
This solution was in workshop to remove details fragment and open the new one